keys/keys.go

```diff
@@ -289,5 +289,11 @@ func DecodeRangeMetaKey(key proto.Key) (proto.Key, proto.Key) {
 	}
 	// Otherwise find the first entry greater than the given key in the same meta prefix.
 	metaPrefix := proto.Key(key[:len(Meta1Prefix)])
+	// Edge case for "\x00\x00meta1\xff\xff", which cannot return
+	// key.Next() as which itself is the last key in Meta1Prefix range.
+	meta1KeyMax := proto.MakeKey(Meta1Prefix, proto.KeyMax)
```

Comment:
On Thu, May 28, 2015 at 11:09 AM es-chow <notifications@github.com> wrote:

> In keys/keys.go
> <https://github.com/cockroachdb/cockroach/pull/1171#discussion_r31243061>:
>
> > @@ -289,5 +289,11 @@ func DecodeRangeMetaKey(key proto.Key) (proto.Key, proto.Key) {
> >  	}
> >  	// Otherwise find the first entry greater than the given key in the same meta prefix.
> >  	metaPrefix := proto.Key(key[:len(Meta1Prefix)])
> > +	// Edge case for "\x00\x00meta1\xff\xff", which cannot return
> > +	// key.Next() as which itself is the last key in Meta1Prefix range.
> > +	meta1KeyMax := proto.MakeKey(Meta1Prefix, proto.KeyMax)
>
>
>    1. Not doing this is that other Max in constants.go like SystemMax,
>    MetaMax greater than the concat(SystemPrefix, \xff\xff), so I am not sure
>    this Meta1KeyMax should define like this. 2 .Current Meta2 is not used in
>    RangeLookup key, it seems no issue.
>
>
Didn't quite follow what you meant. The basic idea is that you want the
constant to be computed once.



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1171/files#r31243061>.
>